diconfig.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+$di = new Aura\Di\Container(new \Aura\Di\Factory());
+
+$di->setAutoResolve(false);
+
+$db = $config['database'];
+
+$dsn = sprintf('mysql:host=%s;dbname=%s', $db['host'], $db['name']);
+
+$di->params['PDO'] = [
+    'dsn' => $dsn,
+    'username' => $db['user'],
+    'passwd' => $db['pass'],
+    'options' => [
+        PDO::ATTR_ERRMODE,
+        PDO::ERRMODE_EXCEPTION,
+    ],
+];
+
+$di->params[Masterclass\Db\Mysql::class] = [
+    'pdo' => $di->lazyNew('PDO'),
+];
+
+$di->params[Masterclass\Request::class] = [
```

Comment:
For services like this I like to set an easy-to-remember key, like:
``
$di->set('request', $di->lazyNew('Masterclass\Request');